spec/services/solidus_pay_tomorrow/client/create_order_service_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe SolidusPayTomorrow::Client::CreateOrderService do
+  let(:order) { create(:order_with_line_items, line_items_count: 2) }
+  let(:payment_method) { create(:pt_payment_method) }
+  let(:http_response) { instance_double(HTTParty::Response, parsed_response: success_response, success?: true) }
+  let(:full_name) { Spree::Address::Name.new(order.bill_address.name) }
+  let(:expected_body) do
+    { orderId: order.number,
+      firstName: full_name.first_name,
+      lastName: full_name.last_name,
+      street: order.bill_address.address1,
+      city: order.bill_address.city,
+      zip: order.bill_address.zipcode,
+      state: order.bill_address.state.abbr,
+      email: order.email,
+      returnUrl: 'https://domain.com/webhooks/return',
+      cancelUrl: 'https://domain.com/webhooks/cancel',
+      notifyUrl: 'https://domain.com/webhooks/notify',
+      cellPhone: order.bill_address.phone,
+      loanAmount: order.total.to_i,
+      applicationItems:
+        [{ description: "As seen on TV!", quantity: 1, price: 10.0, sku: "SKU-1" },
+         { description: "As seen on TV!", quantity: 1, price: 10.0, sku: "SKU-2" }] }.to_json
+  end
+
+  before do
+    url = 'https://api-staging.paytomorrow.com/api/application/ecommerce/orders'
+    headers = { 'Authorization': "Bearer #{ENV['ACCESS_TOKEN']}",
+                'Content-Type': 'application/json' }
+    allow(HTTParty).to receive(:post).with(url, headers: headers, body: expected_body).and_return(http_response)
+  end
+
+  describe '#call' do
+    it 'creates successful order' do
+      response = described_class.call(order: order, payment_method: payment_method)
+      expect(response).to eq(success_response)
+    end
+
+    def success_response
```

Comment:
What is the benefit to having it as a method, rather than making it explicit in the example with:

```
expect(response).to eq(
  url: "https://consumer-staging.paytomorrow.com/verify/personal?app=11171280-8e2a-4b2e-8855-76285fc578c6&auth=e3de6c12-ed9a-4bb1-a066-5c62a1ba19d4",
  token: "11171280-8e2a-4b2e-8855-76285fc578c6"
)
```

Reply:
This is also being used on **Line:8** in the same file.
Hence it makes sense to have it at one place and use it at both these places.

**Variable vs method**
Since this is being used at 2 different places with different scopes, it can only live in a top level let, an instance variable or a method.

1. A let is still a method. And since this is static info, this should not be in let (no need for memoization)
2. Having instance variable is discouraged anyway in specs. Moreover, rubocop complains with - "prefer a local variable, let or a method rather than instance variable"

Hence, method makes sense here. That's why it is here.

Reply:
Fixed as per discussion on call in https://github.com/nebulab/solidus_pay_tomorrow/pull/7/commits/75dc5280c554494a2fd5784ac6c45656b06e9fdc